docs/release-source/release/stubs.md

```diff
@@ -16,36 +16,95 @@ wrapping an existing function with a stub, the original function is not called.
 
 ```javascript
-"test should call all subscribers, even if there are exceptions" : function(){
-    var message = 'an example message';
-    var stub = sinon.stub().throws();
-    var spy1 = sinon.spy();
-    var spy2 = sinon.spy();
+const emailService = require('./emailService');
+
+let orderService = {
+    confirm() {
+        let isEmailSent;
+        try {
+            isEmailSent = emailService.send();
+        } catch (EmailClientError) {
+            return 'Error occurred in email client!';
+        }
 
-    PubSub.subscribe(message, stub);
-    PubSub.subscribe(message, spy1);
-    PubSub.subscribe(message, spy2);
+        if (isEmailSent) return 'Email was sent!';
+        else return 'Internal email service failed!';
+    }
+}
+```
 
-    PubSub.publishSync(message, undefined);
+Without writing the code for its dependency, the 'orderService' can be unit-tested. The only requirement is for 'emailService.js' file to have a 'send' property:
 
-    assert(spy1.called);
-    assert(spy2.called);
-    assert(stub.calledBefore(spy1));
+```javascript
+module.exports = {
+    send() {}
 }
 ```
 
-Note how the stub also implements the spy interface. The test verifies that all
-callbacks were called, and also that the exception throwing stub was called
-before one of the other callbacks.
+ A test can be written for exception handling. To do that, a stub pre-programmed to throw the exception must be provided in place of the dependency's actual function:
+
+```javascript
+describe('orderService', function() {
+    it('should handle EmailClientError when emailService throws', function() {
+
+        // SETUP
+        sinon.stub(emailService, 'send').throws('EmailClientError');
+        // EXERCISE
+        let result = orderService.confirm();
+        // VERIFY
+        expect(result).toBe("Error occurred in email client!");
+    });
+});
+```
+
+A stub can also be provided with the behaviour to return some value. Here, the stub will return true when called, in order to trigger a particular code path within 'orderService':
+
+```javascript
+describe('orderService', function() {
+    it('should return confirmation string when emailService successfully sends', function() {
+
+        // SETUP
+        sinon.stub(emailService, 'send').returns(true);
+        // EXERCISE
+        let result = orderService.confirm();
+        // VERIFY
+        expect(result).toBe("Email was sent!");
+    });
+});
+```
+
+The 2 tests above pass when run in isolation. However, when these tests are run in a single test suite, the second test would fail because the function was already stubbed. In order to change the stub for the same function, the stub needs to be restored:
+
+```javascript
+describe('orderService', function() {
+    it('should handle EmailClientError when emailService throws', function() {
+
+        // SETUP
+        let sendStub = sinon.stub(emailService, 'send').throws('EmailClientError');
```

Comment:
Avoid `let` and other ES2015+ syntax in the examples. We try to have all produced be ES5.1 compatible.